Source/Core/Core/HW/MemmapFunctions.cpp

```diff
@@ -91,9 +92,14 @@ static u32 EFB_Read(const u32 addr)
 
 static void GenerateDSIException(u32 _EffectiveAddress, bool _bWrite);
 
-template <typename T, typename U>
-__forceinline void ReadFromHardware(U &_var, const u32 em_address, Memory::XCheckTLBFlag flag)
+template <XCheckTLBFlag flag, typename T, typename U>
+__forceinline void ReadFromHardware(U &_var, const u32 em_address)
 {
+	int segment = em_address >> 28;
+	// Quick check for an address that can't meet any of the following conditions,
+	// to speed up the MMU path.
+	if (BitSet32(0xCFC)[segment])
+		goto translateaddress;
```

Comment:
There's two ways to get to that code though, so I'm not sure how you'd avoid using a goto? I'll try to see if I can restructure it